onnx/defs/math/defs.cc

```diff
@@ -424,6 +424,37 @@ max(0,x) + min(0,alpha*(exp(x/alpha)-1))
 
 static float celu_default_alpha = 1.0;
 
+TensorProto ToDimensionOneFloatTensor(float value) {
+  auto t = ToTensor(std::vector<float>({value}));
+  t.add_dims(1);
+  return t;
+}
+
+bool BuildContextDependentFunctionBodyCelu(
+    const FunctionBodyBuildContext& ctx,
+    const OpSchema& schema,
+    FunctionProto& functionProto) {
+  std::vector<FunctionBodyHelper::NodeDef> body;
+float alpha = ctx.getAttribute("alpha")->f();
+  body.push_back({{"alpha"},
+                  "Constant",
+                  {},
+                  {MakeAttribute("value", ToDimensionOneFloatTensor(alpha))}});
+
+  body.push_back({{"X_alpha"}, "Div", {"X", "alpha"}});
+  body.push_back({{"Elu_Result"}, "Elu", {"X_alpha"}, {{"alpha", 1.f}}});
```

Comment:
Is that supposed to be 1.f or alpha from above?

Reply:
I think 1.f is the default value for the attribute but we should use celu_default_alpha instead. What do you think?

---
In reply to: [410777439](https://github.com/onnx/onnx/pull/2725#discussion_r410777439) [](ancestors = 410777439)

Reply:
I don't know this op. But looking at the spec, it says the value is
```
   max(0,x) + min(0,alpha*(exp(x/alpha)-1))
```
So, I assume both alphas in this equation must be the attribute-value retrieved above?

Reply:
But I am confused about the "max(0,x)" part too. How does the above equation get rewritten in the form shown in this function-body?

Reply:
Both alphas are attribute-value retrieved above that we put in a constant tensor by the name "alpha" 

---
In reply to: [410779367](https://github.com/onnx/onnx/pull/2725#discussion_r410779367) [](ancestors = 410779367)

Reply:
You make a good point, the test passed so I didn't look too closely but let me think.

---
In reply to: [410779760](https://github.com/onnx/onnx/pull/2725#discussion_r410779760) [](ancestors = 410779760)

Reply:
[https://github.com/onnx/onnx/pull/2575](https://github.com/onnx/onnx/pull/2575) that explains more on the implementation.

---
In reply to: [410779924](https://github.com/onnx/onnx/pull/2725#discussion_r410779924) [](ancestors = 410779924,410779760)

Reply:
``Tensor celu(const Tensor & self, Scalar alpha) {
  double inv_alpha = 1. / alpha.to<double>();
  return at::elu(self, alpha, Scalar(1.0), Scalar(inv_alpha));
}``

---
In reply to: [410780998](https://github.com/onnx/onnx/pull/2725#discussion_r410780998) [](ancestors = 410780998,410779924,410779760)

Reply:
May be this is okay, since there is the multiplication by alpha down below.

Reply:
yep

---
In reply to: [410781381](https://github.com/onnx/onnx/pull/2725#discussion_r410781381) [](ancestors = 410781381)